wandb/sdk/launch/_project_spec.py

```diff
@@ -145,43 +145,68 @@ def get_entry_point(self, entry_point: str) -> "EntryPoint":
             return self._entry_points[entry_point]
         return self.add_entry_point(entry_point)
 
-    def _fetch_project_local(self, api: Api) -> None:
+    def _check_and_download_code_artifacts(self, internal_api: Api) -> bool:
+        public_api = wandb.PublicApi(
+            overrides={"base_url": internal_api.settings("base_url")}
+        )
+
+        run_entity, run_project, run_name = utils.parse_wandb_uri(self.uri)
+        run = public_api.run(f"{run_entity}/{run_project}/{run_name}")
```

Comment:
re: earlier comment especially if we are just going to reconstruct the uri like this lol